fbpic/lpa_utils/laser/transverse_laser_profiles.py

```diff
@@ -0,0 +1,517 @@
+        Chapter 16: Wave optics and Gaussian beams)
+
+        .. warning::
+            The above formula depends on a parameter :math:`m`
+            (see documentation below). In order to be properly resolved by
+            the simulation, a Laguerre-Gauss profile with a given :math:`m`
+            requires the azimuthal modes from :math:`0` to :math:`m+1`.
+            (i.e. the number of required azimuthal modes is ``Nm=m+2``)
+
+            The non-linear plasma response for this profile (e.g.
+            wakefield driven by the ponderomotive force) may require
+            even more azimuthal modes.
+
+        Parameters
+        ----------
+
+        p: int (positive)
+            The order of the Laguerre polynomial. (Increasing ``p`` increases
+            the number of "rings" in the radial intensity profile of the laser.)
+
+        m: int (positive)
+            The azimuthal order of the pulse.
+            (In the transverse plane, the field of the pulse varies as
+            :math:`\cos[m(\\theta-\\theta_0)]`.)
+
+        waist: float (in meter)
+            Laser waist at the focal plane, defined as :math:`w_0` in the
+            above formula.
+
+        zf: float (in meter), optional
+            The position of the focal plane (in the lab frame).
+            If ``zf`` is not provided, the code assumes ``zf=0.``.
+
+        lambda0: float (in meter), optional
+            The wavelength of the laser (in the lab frame), defined as
+            :math:`\\lambda_0` in the above formula.
+            Default: 0.8 microns (Ti:Sapph laser).
+
+        theta0: float (in radian), optional
+            The azimuthal position of (one of) the maxima of intensity, in the
+            transverse plane.
+            (In the transverse plane, the field of the pulse varies as
+            :math:`\cos[m(\\theta-\\theta_0)]`.)
+
+        propagation_direction: int, optional
+            Indicates in which direction the laser propagates.
+            This should be either 1 (laser propagates towards positive z)
+            or -1 (laser propagates towards negative z).
+        """
+        # Initialize propagation direction
+        LaserTransverseProfile.__init__(self, propagation_direction)
+
+        # Wavevector and Rayleigh length
+        k0 = 2*np.pi/lambda0
+        zr = 0.5*k0*waist**2
+        # Scaling factor, so that the pulse energy is independent of p and m.
+        if m ==0:
+            scaled_amplitude = 1.
+        else:
+            scaled_amplitude = np.sqrt( factorial(p)/factorial(m+p) )
+        if m != 0:
+            scaled_amplitude *= 2**.5
+        # Store the parameters
+        if m < 0 or type(m) is not int:
+            raise ValueError("m should be an integer positive number.")
+        self.p = p
+        self.m = m
+        self.scaled_amplitude = scaled_amplitude
+        self.laguerre_pm = genlaguerre(self.p, self.m) # Laguerre polynomial
+        self.theta0 = theta0
+        self.k0 = k0
```

Comment:
It seems that we do not need to store `k0`.
```suggestion
        self.k0 = k0
```